tests/utils/test_commands.py

```
@@ -0,0 +1,90 @@
+"""
+Tests for command module
+"""
+from subprocess import CalledProcessError
+
+import pytest
+from cg.utils import Process
+
+
+def test_process():
+    # GIVEN a binary with 'ls'
+    binary = "ls"
+    # WHEN instantiating a Process with the binary
+    process = Process(binary=binary)
+    # THEN assert the instatiation worked as expected
+    assert process.binary == binary
+    assert repr(process) == "Process:base_call:['ls']"
+    assert process.stdout == ""
+
+
+def test_process_run_invalid_command():
+    # GIVEN a binary with 'ls'
+    binary = "false"
+    # WHEN instantiating a Process with the binary
+    process = Process(binary=binary)
+    # THEN assert the instatiation worked as expected
+    with pytest.raises(CalledProcessError):
+        process.run_command()
+
+
+def test_process_run_command_no_params(ls_process):
+    # GIVEN a proces with 'ls' as binary
+    process = ls_process
+    # WHEN running the command without parameters
+    res = process.run_command()
+    # THEN assert it returns valid exit code
+    assert res == 0
+    # THEN assert the output is valid
+    assert "setup.py" in process.stdout
+
+
+def test_process_run_command_with_params(ls_process):
+    # GIVEN a proces with 'ls' as binary
+    process = ls_process
+    # WHEN running the command with parameters
+    res = process.run_command(["-l"])
+    # THEN assert it returns valid exit code
+    assert res == 0
+    # THEN assert the output is valid
+    i = 0
+    for i, line in enumerate(process.stdout.split("\n"), 1):
+        if i > 1:
+            assert len(line.split()) == 9
```

seems like we expect a certain number of files? could we make this more robust?

---

also having an assert in an if might hide errors if the condition is unfulfilled

---

No we are testing a method on the Process class